syntax = "proto3";
package raylet;

import public "volpy.proto";

message WorkerData {
  string port = 1; // It's easier to deal with str when we don't need input validation here
}

message AllTasks {
  repeated TaskNameAndData all_tasks = 1;
}

service Volpy {
  // Worker
  rpc InitWorker(WorkerData) returns (Status) {}

  // Task
  rpc CreateTask(TaskNameAndData) returns (Status) {}
  rpc SubmitTask(IdTaskArgs) returns (StatusWithDataRef) {}
  rpc GetAllTasks(Empty) returns (AllTasks) {}

  // DataStore
  rpc Get(DataRef) returns (StatusWithData) {}
  rpc Put(Data) returns (StatusWithDataRef) {}
}

